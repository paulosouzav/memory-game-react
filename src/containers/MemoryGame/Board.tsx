import styled, { css } from 'styled-components';

interface Props {
  large?: boolean;
  medium?: boolean;
}

const Board = styled.div`
  width: 100%;
  max-width: 720px;
  margin: auto;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  perspective: 1000px;

  ${({ medium }: Props) => medium && MEDIUM}
  ${({ large }: Props) => large && LARGE}
`;

const MEDIUM = css`
  max-width: 1050px;
`;

const LARGE = css`
  max-width: unset;
`;

Board.displayName = 'Board';

export default Board;
