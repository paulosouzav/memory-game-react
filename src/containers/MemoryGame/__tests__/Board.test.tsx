import React from 'react';
import Board from '../Board';
import { mount } from 'enzyme';
import 'jest-styled-components';

describe('Board', () => {
  it('should render without failing', () => {
    const wrapper = mount(<Board />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should render medium Board', () => {
    const wrapper = mount(<Board medium />);
    expect(wrapper).toHaveStyleRule('max-width', '1050px');
    expect(wrapper).toMatchSnapshot();
  });

  it('should render large Board', () => {
    const wrapper = mount(<Board large />);
    expect(wrapper).toHaveStyleRule('max-width', 'unset');
    expect(wrapper).toMatchSnapshot();
  });
});
