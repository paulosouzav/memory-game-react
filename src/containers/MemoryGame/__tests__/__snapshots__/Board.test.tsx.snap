// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Board should render large Board 1`] = `
.c0 {
  width: 100%;
  max-width: 720px;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
  max-width: unset;
}

<Board
  large={true}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  width: 100%;
  max-width: 720px;
  margin: auto;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  perspective: 1000px;

  ",
            [Function],
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "Board",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    large={true}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</Board>
`;

exports[`Board should render medium Board 1`] = `
.c0 {
  width: 100%;
  max-width: 720px;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
  max-width: 1050px;
}

<Board
  medium={true}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  width: 100%;
  max-width: 720px;
  margin: auto;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  perspective: 1000px;

  ",
            [Function],
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "Board",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    medium={true}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</Board>
`;

exports[`Board should render without failing 1`] = `
.c0 {
  width: 100%;
  max-width: 720px;
  margin: auto;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-perspective: 1000px;
  -moz-perspective: 1000px;
  -ms-perspective: 1000px;
  perspective: 1000px;
}

<Board>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "sc-bdVaJa",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "
  width: 100%;
  max-width: 720px;
  margin: auto;
  display: flex;
  flex-wrap: wrap;
  align-items: center;
  justify-content: center;
  perspective: 1000px;

  ",
            [Function],
            "
  ",
            [Function],
            "
",
          ],
        },
        "displayName": "Board",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "sc-bdVaJa",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
  >
    <div
      className="c0"
    />
  </StyledComponent>
</Board>
`;
